### PR TITLE
State can add certifing actuary on rate details page and additional actuary on contacts page

### PR DESCRIPTION
**Pull request by haworku**
## Summary
State can add certifying actuary on rate details page
State can only add additional actuary on contact page
Certifying actuary contact moved to rate details section on review/submit and summary pages.

#### Related issues
Jira: 

- [MR-2238](https://qmacbis.atlassian.net/browse/MR-2238)
- [MR-2233](https://qmacbis.atlassian.net/browse/MR-2233)

Contains work from previous PRs:

- https://github.com/CMSgov/managed-care-review/pull/1288 
- https://github.com/CMSgov/managed-care-review/pull/1276

#### Test cases covered
Unit Tests:
- RateDetails.test.tsx
   - `handles multiple rates`: Update tests with assertions for certifying actuary contact fields

- healthPlanFormData.test.ts:
   - `'hasValidRates evaluates as expected'`: Update tests to cover missing actuary contact fields

- toLatestProtoVersion.test.ts
   - `'should convert actuary contacts after the first to addtlActuaryContacts and communication preference as expected'`
   - `'should return empty array for addtlActuaryContacts when only a single actuary contact exists'`

- Contacts.test.tsx
   - `'text renders correctly when multi-rate-submissions flag is turned on'`
   
- ContactsSummarySection.test.tsx
   - `'renders correct text for actuary contact fields when multi-rate-submissions flag is on'`
   
- UploadedDocumentsTable.test.tsx
   - Fixed passing tests that logged errors
   
- SupportingDocumentsSummarySection.test.tsx
   - Fixed passing tests that logged errors

Cypress Tests:
- contacts.spec.ts
   - `'can navigate to and from contacts page with contract and rates submission with multi-rate-submissions flag on'`
   - `fillOutNewRateCertification` and `fillOutAmendmentToPriorRateCertification`: Code to fill out certifying actuary contact fields

### CODE REVIEW

File: services/app-web/src/pages/StateSubmission/Contacts/Contacts.tsx [hunk 1]
```diff
@@ -219,10 +220,27 @@ export const Contacts = ({
...
+    /**
+     * Multi-rate-submissions flag off:
+     * Certifying actuary is displayed and fill out on this page so we need to display it along with additional actuaries
+     * when flag is off.
+     *
+     * We do this by combining the certifying actuary of the first-rate certification with the actuaries in
+     * addtlActuaryContacts into one array.
+     */
+    const combineActuaries = () => {
+        if (draftSubmission.rateInfos[0]?.actuaryContacts.length) {
```
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/Contacts/Contacts.tsx:232]: Yep, this makes sense. Good catch

**Comment by JasonLin0991** [comment]
@haworku, @rswerve merged in his work on the summary page. I'll be merging main in again with his changes and make updates to the summary page for the actuary work, specifically on rate details section.

File: services/app-web/src/pages/StateSubmission/Contacts/Contacts.tsx [hunk 2]
```diff
@@ -266,9 +301,33 @@ export const Contacts = ({
...
+         * Multi-rate-submissions flag off:
+         * Certifying actuary is on this page along with additional actuaries in one array, values.actuaryContacts from
+         * formik. In v2 of the protobuf, it was directly saved into the first-rate certification actuaryContacts field.
+         * So we need to now save this data that conforms to v3 of the protobuf where certifying actuary is saved to each
+         * rate certification data and the additional actuaries to on the top level in addtlActuaryContacts.
+         *
+         * We can do that by saving the first actuary (the certifying actuary) in values.actuaryContacts value from
+         * formik, to the first-rate certifications actuaryContacts at index 0, which is the first actuary contact. Then
+         * we save the rest of the actuary contacts in values.actuaryContacts to the addtlActuaryContacts field
+         */
```
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/Contacts/Contacts.tsx:315]: Here is a simpler comment I would suggest. I would rather not have comments about proto versions in React code, now with toLatestVersion the React UI only needs to concern itself with the current proto. 

 All issues with proto versions should be handled in the deserialization step, isolating the React frontend code from these concerns about the shape or values in the proto over time. 

```suggestion
         * Multi-rate-submissions flag on: formik values.actuaryContacts are saved as additional actuaryContacts
         * Multi-rate-submissions flag off: first formik values.actuaryContacts is saved as the certifying actuary on the first rate info. Any other  values.actuaryContacts are saved as additional actuary contacts
         *
         */
```

**Comment by haworku** [comment]
Everything looks good here to me. Excited to get this in the hands of users soon and I like that removing the flag looks like it should be pretty reasonable to do as well.  Approved! (i can't approve my own PR)

**Review by JasonLin0991 — APPROVED** [review]

**Pull request merged**